comit/src/expiries.rs

```diff
@@ -0,0 +1,591 @@
+    fn next(&self) -> (AliceAction, AliceState) {
+        use self::{AliceAction::*, AliceState::*};
+
+        match self {
+            None => (Create, Created),
+            Created => (FundAlpha, FundAlphaTransactionSent),
+            FundAlphaTransactionSent => (WaitForFundTransactionFinality, AlphaFunded),
+            AlphaFunded => (WaitForBetaFunded, BetaFunded),
+            BetaFunded => (RedeemBeta, RedeemBetaTransactionSent),
+            RedeemBetaTransactionSent => (WaitForRedeemBetaTransactionFinality, Done),
+            Done => (NoFurtherAction, Done),
+        }
+    }
+
+    /// The minimum time we need to allow to transition to the next state.
+    fn transition_period(&self, c: &Config) -> Duration {
+        use self::AliceAction::*;
+
+        let (next_action, _next_state) = self.next();
+
+        match next_action {
+            Create => {
+                // Transition from None to Created
+                c.alice_to_act()
+            }
+            FundAlpha => {
+                // Transition from Created to FundAlphaTransactionSent
+                c.alice_to_act()
+            }
+            WaitForFundTransactionFinality => {
+                // Transition from FundAlphaTransactionSent to AlphaFunded
+                c.include_alpha_fund_transaction() + c.finality_alpha()
+            }
+            WaitForBetaFunded => {
+                // Transition from AlphaFunded to BetaFunded
+                c.bob_to_act() + c.include_beta_fund_transaction() + c.finality_beta()
+            }
+            RedeemBeta => {
+                // Transition from BetaFunded to RedeemBetaTransactionSent
+                c.alice_to_act()
+            }
+            WaitForRedeemBetaTransactionFinality => {
+                // Transition from RedeemBetaTransactionSent to Done
+                c.include_beta_redeem_transaction() + c.finality_beta()
+            }
+            // Transitioning to any of the cancel actions takes, be definition, zero time.
+            NoFurtherAction | Abort | WaitForRefund | Refund => Duration::zero(),
+        }
+    }
+
+    /// True if Alice has funded the alpha HTLC. We return true as soon as the
+    /// fund transaction has been sent since any time after attempting to refund
+    /// is the correct cancellation action.
+    fn has_funded_alpha(&self) -> bool {
+        match self {
+            AliceState::None | AliceState::Created => false,
+            AliceState::FundAlphaTransactionSent
+            | AliceState::AlphaFunded
+            | AliceState::BetaFunded
+            | AliceState::RedeemBetaTransactionSent
+            | AliceState::Done => true,
+        }
+    }
+}
+
+/// Happy path states for Bob.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum BobState {
+    // Bob does not have a None state because swap creation time is implicit in Bob's first time
+    // to act after Alice funds.
+    /// Initial state, swap has been created.
```

Comment:
Again, I feel like `Started` might be better. `swap has been created.` created where?